### PR TITLE
Crop Mirror Normalize Tensor implementation

### PR DESCRIPTION
**Pull request by r-abishek**
- Adds crop mirror normalize tensor implementation on host and hip
- Adds corresponding unit and performance tests

### CODE REVIEW

File: src/include/hip/rpp_hip_common.hpp [hunk 1]
```diff
@@ -497,13 +514,28 @@ __device__ __forceinline__ void rpp_hip_load8_and_unpack_to_float8(uchar *srcPtr
     src_f8->y = rpp_hip_unpack(src.y);
 }
 
+__device__ __forceinline__ void rpp_hip_load8_and_unpack_to_float8_mirror(uchar *srcPtr, int srcIdx, d_float8 *src_f8)
```
Comment (rrawther) [inline src/include/hip/rpp_hip_common.hpp:517]: probably you can pass (srcPtr + srcIdx) directly to this function (remove srcIdx), applies to all such calls

**Review by rrawther — CHANGES_REQUESTED** [review]
please address review comments

**Comment by paveltc** [comment]
@rrawther @asalmanp @r-abishek Is this PR ready for unit tests or are more changes going to be made to it?

**Comment by r-abishek** [comment]
> @rrawther @asalmanp @r-abishek Is this PR ready for unit tests or are more changes going to be made to it?

@paveltc A few changes from the review comments are pending. I'll leave a comment once we have them done.

**Comment by paveltc** [comment]
@r-abishek Are the changes still pending?

**Comment by kiritigowda** [comment]
@r-abishek can you fix the conflicts to run the CI?

File: src/include/hip/rpp_hip_common.hpp (hunk 1 above)
Comment (r-abishek) [reply src/include/hip/rpp_hip_common.hpp]: Done

**Comment by r-abishek** [comment]
@kiritigowda Merge conflicts should be fixed now.
@paveltc A couple changes have been made to this PR now. Calls to some of the common helper functions were changed:
- CMN host and hip are the new additions in the PR.
- But the common helper functions changed are also used by some other HIP functionalities like blend, box filter, erode, dilate, color cast, color twist.

**Comment by paveltc** [comment]
@rrawther @asalmanp @r-abishek I will run the unit tests on this PR today and comment when I am done.

**Review by rrawther — APPROVED** [review]
@paveltc : make sure unit tests passing before merging

**Comment by paveltc** [comment]
@rrawther @r-abishek @asalmanp 
All of the relevant unit tests for this PR pass.
These include:
Crop Mirror Normalize: HOST and HIP
Blend: HIP
Box Filter: HIP
Color Twist : HIP
Grid Mask : HIP
Exposure: HIP

**Pull request merged**